src/display/api.js

```diff
@@ -1574,32 +1594,72 @@ var WorkerTransport = (function WorkerTransportClosure() {
       var messageHandler = this.messageHandler;
       var loadingTask = this.loadingTask;
 
-      var pdfDataRangeTransport = this.pdfDataRangeTransport;
-      if (pdfDataRangeTransport) {
-        pdfDataRangeTransport.addRangeListener(function(begin, chunk) {
-          messageHandler.send('OnDataRange', {
-            begin,
-            chunk,
-          });
-        });
+      messageHandler.on('GetReader', function(data, sink) {
+        this._fullReader = this._networkStream.getFullReader();
 
-        pdfDataRangeTransport.addProgressListener(function(loaded) {
-          messageHandler.send('OnDataProgress', {
-            loaded,
-          });
-        });
+        sink.onPull = () => {
+          this._fullReader.read().then(function({ value, done, }) {
+            if (done) {
+              sink.close();
+              return;
+            }
+            // Enqueue data chunk into sink, and transfer it
+            // to other side as `Transferable` object.
+            sink.enqueue(new Uint8Array(value), 1, [value]);
+          }).catch(sink.error);
+        };
+
+        sink.onCancel = (reason) => {
+          this._fullReader.cancel(reason);
+        };
+      }, this);
+
+      messageHandler.on('ReaderHeadersReady', function(data) {
+        let headersCapability = createPromiseCapability();
+        let fullReader = this._fullReader;
+
+        fullReader.headersReady.then(() => {
+          if (!fullReader.isStreamingSupported ||
+              !fullReader.isRangeSupported) {
```

Comment:
Nit: You can probably reduced the overall indentation a bit by using an early return here, i.e.
```javascript
  if (fullReader.isStreamingSupported && fullReader.isRangeSupported) {
    return;
  }
  fullReader.onProgress = (evt) => {
    ...
```